claim_batch/services.py

```diff
@@ -468,43 +468,53 @@ def do_process_batch(audit_user_id, location_id, period, year):
                 .filter(valid_from__gte = end_date)\
                 .filter(product = product)
             # 1.3 retrive all the Item & service per product
+            # to be checked if we need to pull the claim too
             work_data.items  = ClaimItem.objects\
                 .filter(validity_from__lte = start_date)\
                 .filter(validity_from__gte = end_date)\
                 .filter(validity_to__isnull = True)\
                 .filter(claim_pocess_stamp_lte = end_date  )\
                 .filter(claim_pocess_stamp_gte = end_date  )\
-                .filter(product = product)
+                .filter(product = product)\
+                .select_related('claim__health_facility')\
+                .orderby('claim__health_facility').orderby('claim_id')
             work_data.services  = ClaimService.objects\
                 .filter(validity_from__lte = start_date)\
                 .filter(validity_from__gte = end_date)\
                 .filter(claim_pocess_stamp_lte = end_date  )\
                 .filter(claim_pocess_stamp_gte = end_date  )\
                 .filter(validity_to__isnull = True)\
-                .filter(product = product)
+                .filter(product = product)\
+                .select_related('claim__health_facility')\
+                .orderby('claim__health_facility').orderby('claim_id')
             # 1.4 retrive all contributions per product // process_stamp should be the creation date
             work_data.contributions  = Premium.objects\
                 .filter(validity_from__lte = start_date)\
                 .filter(validity_from__gte = end_date)\
                 .filter(pocess_stamp_lte = end_date  )\
```

Comment:
@delcroip Does this field exist in 'Premium' object. I cannot see such name of field in this model.

Reply:
you are right pocess_stamp must be added on the database and the date when the Premium is saved in the database (managed by the SQL server)

Reply:
ok noted